pkg/util/term/term.go

```diff
@@ -60,17 +60,17 @@ type fd interface {
 // IsTerminalIn returns true if t.In is a terminal. Does not check /dev/tty
 // even if TryDev is set.
 func (t TTY) IsTerminalIn() bool {
-	return isTerminal(t.In)
+	return IsTerminal(t.In)
 }
 
 // IsTerminalOut returns true if t.Out is a terminal. Does not check /dev/tty
 // even if TryDev is set.
 func (t TTY) IsTerminalOut() bool {
-	return isTerminal(t.Out)
+	return IsTerminal(t.Out)
 }
 
-// isTerminal returns whether the passed object is a terminal or not
-func isTerminal(i interface{}) bool {
+// IsTerminal returns whether the passed object is a terminal or not
+func IsTerminal(i interface{}) bool {
```

Comment:
This is how you the code was before. It's used by OpenShift and we pass in
things like io.Reader. Take a look at where we use it downstream.

On Friday, July 15, 2016, Dr. Stefan Schimanski <notifications@github.com>
wrote:

> In pkg/util/term/term.go
> <https://github.com/kubernetes/kubernetes/pull/28964#discussion_r70931829>
> :
>
> >  }
> >
> > -// isTerminal returns whether the passed object is a terminal or not
> > -func isTerminal(i interface{}) bool {
> > +// IsTerminal returns whether the passed object is a terminal or not
> > +func IsTerminal(i interface{}) bool {
>
> Don't like the signature for a public function. How about making fd
> public above and then use func IsTerminal(fd Fd) bool?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/28964/files/792ec603c0ac6891c4c547ce4cbefcc3101503d4#r70931829>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAABYlZKHNW8iAer9mIjW5mi-k6-iZsTks5qVzb9gaJpZM4JMn10>
> .
>